apps/omg_watcher/lib/omg_watcher/block_getter.ex

```diff
@@ -66,18 +66,11 @@ defmodule OMG.Watcher.BlockGetter do
     GenServer.start_link(__MODULE__, args, name: __MODULE__)
   end
 
-  @doc """
-  Initializes the GenServer state, most work done in `handle_continue/2`.
-  """
-  def init(args) do
-    {:ok, args, {:continue, :setup}}
-  end
-
   @doc """
   Reads the status of block getting and application from `OMG.DB`, reads the current state of the contract and root
   chain and starts the pollers that will take care of getting blocks.
   """
-  def handle_continue(:setup, args) do
+  def init(args) do
```

Comment:
Same here about handle_continue, especially `ExitProcessor.check_validity`, `OMG.DB.get_single_value`, `check_in_to_coordinator ` and things that could hide behind all those functions (nothing intensive from what I can see but they jump through a lot of modules and functions).